task.sql

```diff
@@ -1,11 +1,56 @@
--- Use our database
-USE ShopDB; 
+DROP DATABASE IF EXISTS ShopDB;
+CREATE DATABASE ShopDB;
+USE ShopDB;
 
--- Some data should be created outside the transaction (here)
+CREATE TABLE Products (
+                          ID INT AUTO_INCREMENT,
+                          Name VARCHAR(50),
+                          Description VARCHAR(100),
+                          Price INT,
+                          WarehouseAmount INT,
+                          PRIMARY KEY (ID)
+);
 
--- Start the transaction 
-START TRANSACTION; 
+CREATE TABLE Customers (
+                           ID INT AUTO_INCREMENT,
+                           FirstName VARCHAR(50),
+                           LastName VARCHAR(50),
+                           Email VARCHAR(50),
+                           Address VARCHAR(50),
+                           PRIMARY KEY (ID)
+);
+
+CREATE TABLE Orders (
+                        ID INT AUTO_INCREMENT,
+                        CustomerID INT,
+                        FOREIGN KEY (CustomerID) REFERENCES Customers(ID) ON DELETE SET NULL,
+                        Date DATE,
+                        PRIMARY KEY (ID)
+);
+
+CREATE TABLE OrderItems (
+                            ID INT AUTO_INCREMENT,
+                            OrderID INT,
+                            FOREIGN KEY (OrderID) REFERENCES Orders(ID) ON DELETE SET NULL,
+                            ProductID INT,
+                            Count INT,
+                            FOREIGN KEY (ProductID) REFERENCES Products(ID) ON DELETE SET NULL,
+                            PRIMARY KEY (ID)
+);
+
+INSERT INTO Products (Name, Description, Price, WarehouseAmount)
+VALUES ('AwersomeProduct', 'Product Desctiption', 5, 42);
```

Comment:
There is a typo in the product name 'AwersomeProduct'. It should likely be 'AwesomeProduct'.